### DISCUSSION TITLE
how to use the same dynamic page for creating new - and editing existing items

### DISCUSSION THREAD

**Discussion by Roland-Hufnagel** [neuefische/questions#288] [Web]
To solve this problem I use a dynamic page with SSR and look for the dynamic string (coming from the url) if it is 'new' or not and create either new (empty) props or get the props from the DB. 

I am not sure, if this is a good solution or is there a better way to solve the problem?

``` 
export async function getServerSideProps(context) {
  const { id } = context.params;
  const result = {
    props: {
      title: "",
      questions: [{ title: "", type: "" }],
    },
  };
  if (id !== "new") {
    const survey = getSurveyById(id);
    result.props = { ...survey };
  }
  return result;
} 
```

the link to create new Items: `<Link href="/create/new" passHref>`

**Comment by ThomasF85** [comment]
My guess is that you don't want to create two almost identical pages in your app and that's why you want to distinguish between creating a new item and editing an existing item here.
My guess is that you can take the component that allows for editing of the item (propably a page with some state, functions and a form) and turn it into a reusable component. You can then create two pages that basically just use this component. Your URLs could then also be more semantic, for example: "/create" and "edit/[id]". Your create page would also not need any server side props then.
